src/controllers/TalksController.php

```diff
@@ -498,12 +498,85 @@ protected function getTalkDataFromRequest(PDO $db, Request $request, $event_id)
             FILTER_SANITIZE_URL
         );
 
-        $talk['speakers'] = array_map(function ($speaker) {
-            $speaker = filter_var($speaker, FILTER_SANITIZE_STRING);
-            $speaker = trim($speaker);
-            return $speaker;
-        }, (array) $request->getParameter('speakers'));
+        $talk['speakers'] = array_map(
+            function ($speaker) {
+                $speaker = filter_var($speaker, FILTER_SANITIZE_STRING);
+                $speaker = trim($speaker);
+                return $speaker;
+            },
+            (array) $request->getParameter('speakers')
+        );
+
+        return $talk;
+    }
+
+    public function getSpeakersForTalk(Request $request, PDO $db)
+    {
+        $talk_id = $this->getItemId($request);
+        $talk = $this->getTalkById($db, $request, $talk_id);
+        return $talk->speakers;
+    }
+
+    public function setSpeakerForTalk(Request $request, PDO $db)
```

Comment:
I understand what you are saying, the but the API design does lend itself to 'fat controllers' rather - see https://github.com/joindin/joindin-api/blob/master/src/controllers/TalksController.php#L48 as an example, or https://github.com/joindin/joindin-api/blob/master/src/controllers/EventsController.php#L149